tests/compilers/wast.rs

```diff
@@ -45,7 +45,14 @@ pub fn run_wast(wast_path: &str, compiler: &str) -> anyhow::Result<()> {
         "Running wast `{}` with the {} compiler",
         wast_path, compiler
     );
-    let try_nan_canonicalization = wast_path.contains("nan-canonicalization");
+
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "test-llvm")] {
+            let try_nan_canonicalization = wast_path.contains("nan-canonicalization") || wast_path.contains("f32") || wast_path.contains("f64") || wast_path.contains("float");
+        } else {
+            let try_nan_canonicalization = wast_path.contains("nan-canonicalization");
+        }
+    }
```

Comment:
I think all compilers should behave the same way if a feature X is enabled or disabled.
So I don't think is a good idea to change the features depending on the compiler used (other than support or not support something)